tests/postgres/select_test.go

```diff
@@ -920,6 +921,136 @@ FROM dvds.actor
+     "rFilms"."film.rating" AS "film.rating"
+FROM dvds.actor
+     INNER JOIN dvds.film_actor ON (actor.actor_id = film_actor.film_id)
+     INNER JOIN LATERAL (
+          SELECT film.film_id AS "film.film_id",
+               film.title AS "film.title",
+               film.rating AS "film.rating"
+          FROM dvds.film
+          WHERE (film.rating = 'R') AND (film.film_id = film_actor.film_id)
+     ) AS "rFilms" ON TRUE;
+`
+
+	rRatingFilms := LATERAL(
+		Film.
+			SELECT(
+				Film.FilmID,
+				Film.Title,
+				Film.Rating,
+			).
+			WHERE(Film.Rating.EQ(enum.MpaaRating.R).AND(Film.FilmID.EQ(FilmActor.FilmID)))).
+		As("rFilms")
+
+	query := Actor.
+		INNER_JOIN(FilmActor, Actor.ActorID.EQ(FilmActor.FilmID)).
+		INNER_JOIN(rRatingFilms, Bool(true)).
+		SELECT(
+			Actor.AllColumns,
+			FilmActor.AllColumns,
+			rRatingFilms.AllColumns(),
+		)
+	testutils.AssertDebugStatementSql(t, query, expectedQuery)
+}
+
+func TestCrossJoinLateralQuery(t *testing.T) {
+	expectedQuery := `
+SELECT actor.actor_id AS "actor.actor_id",
+     actor.first_name AS "actor.first_name",
+     actor.last_name AS "actor.last_name",
+     actor.last_update AS "actor.last_update",
+     film_actor.actor_id AS "film_actor.actor_id",
+     film_actor.film_id AS "film_actor.film_id",
+     film_actor.last_update AS "film_actor.last_update",
+     "rFilms"."film.film_id" AS "film.film_id",
+     "rFilms"."film.title" AS "film.title",
+     "rFilms"."film.rating" AS "film.rating"
+FROM dvds.actor
+     INNER JOIN dvds.film_actor ON (actor.actor_id = film_actor.film_id)
+     CROSS JOIN LATERAL (
+          SELECT film.film_id AS "film.film_id",
+               film.title AS "film.title",
+               film.rating AS "film.rating"
+          FROM dvds.film
+          WHERE (film.rating = 'R') AND (film.film_id = film_actor.film_id)
+     ) AS "rFilms";
+`
+
+	rRatingFilms := LATERAL(Film.
+		SELECT(
+			Film.FilmID,
+			Film.Title,
+			Film.Rating,
+		).
+		WHERE(Film.Rating.EQ(enum.MpaaRating.R).AND(Film.FilmID.EQ(FilmActor.FilmID)))).
+		As("rFilms")
+
+	query := Actor.
+		INNER_JOIN(FilmActor, Actor.ActorID.EQ(FilmActor.FilmID)).
+		CROSS_JOIN(rRatingFilms).
+		SELECT(
+			Actor.AllColumns,
+			FilmActor.AllColumns,
+			rRatingFilms.AllColumns(),
+		)
+	testutils.AssertDebugStatementSql(t, query, expectedQuery)
+}
+
+func TestLateralQuery(t *testing.T) {
+	expectedQuery := `
+SELECT actor.actor_id AS "actor.actor_id",
+     actor.first_name AS "actor.first_name",
+     actor.last_name AS "actor.last_name",
+     actor.last_update AS "actor.last_update",
+     film_actor.actor_id AS "film_actor.actor_id",
+     film_actor.film_id AS "film_actor.film_id",
+     film_actor.last_update AS "film_actor.last_update",
+     "rFilms"."film.film_id" AS "film.film_id",
+     "rFilms"."film.title" AS "film.title",
+     "rFilms"."film.rating" AS "film.rating"
+FROM dvds.actor
+     INNER JOIN dvds.film_actor ON (actor.actor_id = film_actor.film_id)
+     LATERAL (
```

Comment:
There is no such thing as LATERAL join. When there is no join specified, it will default to CROSS JOIN. 
This is just a CROSS JOIN between two tables, where the second is lateral query. So this will be equivalent to:
```go
.CROSS_JOIN(LATERAL(....).AS("rFilms"))
```
Also you are missing a comma here.

Reply:
LATERAL are not joins and not intended to use with JOINS only, LATERAL are for subqueries, according to documentation:

>Subqueries appearing in FROM can be preceded by the key word LATERAL...
> A LATERAL item can appear at top level in the FROM list, or within a JOIN tree. In the latter case it can also refer to any items that are on the left-hand side of a JOIN that it is on the right-hand side of.

LATERAL dont defaults to CROSS JOIN.

Reply:
What I wanted to say, if there was a comma (at the end of line 1024) it would be CROSS JOIN. Written this way, I don't know what LATERAL represents.

According to documentation: 
From clause is in the form:
`FROM table_reference [, table_reference [, ...]]` -- top level from list

A table reference can be a table name (possibly schema-qualified), or a derived table such as a subquery, a JOIN construct, or complex combinations of these.

Where join table is this form:
`T1 join_type T2 [ join_condition ]`

In this example. T1 is `dvds.actor`, T2 is `dvds.film_actor` and join condition is `ON (actor.actor_id = film_actor.film_id)`.

What is LATERAL? Is it part of the `join_condition`?
And we know it is not a join type.

Reply:
LATERAL represents a subquery, and yes it's missing a comma, all LATERAL subqueries must be separated by commas